src/mpid/ch4/netmod/ofi/ofi_am_impl.h

```diff
@@ -181,6 +182,26 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_OFI_progress_do_queue(int vni_idx)
     goto fn_exit;
 }
 
+MPL_STATIC_INLINE_PREFIX int MPIDI_OFI_progress_poll(int vni)
+{
+    int mpi_errno = MPI_SUCCESS;
+    int ret;
+    MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_OFI_PROGRESS_POLL);
+    MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_OFI_PROGRESS_POLL);
+
+    void *cq_with_event;
+    ret = fi_poll(MPIDI_OFI_global.ctx[vni].pollset, &cq_with_event, 1);
+    if (unlikely(ret < 0 && ret != -FI_EAGAIN)) {
```

Comment:
Looks like `cq_with_event` is set but never used. I have to check the [fi_poll definition ](https://ofiwg.github.io/libfabric/v1.11.1/man/fi_poll.3.html)to understand how it is used.

Can you add a comment here saying that  `cq_with_event` merely describes new events in completion queues and counters, the consumption of such events are still handled by the corresponding event reading functions (e.g., cq_read, cntr_read|wait)?

Reply:
I believe the intended libfabric api is like this:
```
    fi_poll_open(pollset,...);
    /* add multiple cq, cntr */
    fi_poll_add(cq1,...)
    fi_poll_add(cq2,...)
    fi_poll_add(cntr1,...)
    fi_poll_add(cntr2,...)
```
Then poll multiple cq, cntr in a single call:
```
    void *contexts[MAX_NUM_CTX];
    num_with_events = fi_poll(pollset, ...)
    for (i = 0; i < num_with_events; i++) {
        if (contexts[i] == cq1) {
            fi_cq_read(...)
            /* process cq entries */
        } else if (contexts[i] == cntr1) {
            fi_cntr_read(...)
        } else ... {
        }
    }
```
Because we will have regular progress loop calling `fi_cq_read` and `fi_cntr_read` anyway, here we are just hacking it to poke libfabric progress.